pkg/diagnostics/cluster/aggregated_logging/deploymentconfigs.go

```diff
@@ -0,0 +1,128 @@
+package aggregated_logging
+
+import (
+	"fmt"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/selection"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+const (
+	componentNameEs        = "es"
+	componentNameEsOps     = "es-ops"
+	componentNameKibana    = "kibana"
+	componentNameKibanaOps = "kibana-ops"
+	componentNameCurator   = "curator"
+)
+
+// loggingComponents are those 'managed' by rep controllers (e.g. fluentd is deployed with a DaemonSet)
+var loggingComponents = sets.NewString(componentNameEs, componentNameEsOps, componentNameKibana, componentNameKibanaOps, componentNameCurator)
+
+const deploymentConfigWarnMissingForOps = `
+Did not find a DeploymentConfig to support component '%s'.  If you require
+a separate ElasticSearch cluster to aggregate operations logs, please re-install
+or update logging and specify the appropriate switch to enable the ops cluster.
+`
+
+const deploymentConfigZeroPodsFound = `
+There were no Pods found that support logging.  Try running
+the following commands for additional information:
+
+  oc describe dc -n %[1]s
+  oc get events -n %[1]s
+`
+const deploymentConfigNoPodsFound = `
+There were no Pods found for DeploymentConfig '%[1]s'.  Try running
+the following commands for additional information:
+
+  oc describe dc %[1]s -n %[2]s
+  oc get events -n %[2]s
+`
+const deploymentConfigPodsNotRunning = `
+The Pod '%[1]s' matched by DeploymentConfig '%[2]s' is not in '%[3]s' status: %[4]s. 
+
+Depending upon the state, this could mean there is an error running the image 
+for one or more pod containers, the node could be pulling images, etc.  Try running
+the following commands for additional information:
+
+  oc describe pod %[1]s -n %[5]s
+  oc logs %[1]s -n %[5]s
+  oc get events -n %[5]s
+`
+
+func checkDeploymentConfigs(r diagnosticReporter, adapter deploymentConfigAdapter, project string) {
+	req, _ := labels.NewRequirement(loggingInfraKey, selection.Exists, nil)
+	selector := labels.NewSelector().Add(*req)
+	r.Info("AGL0040", fmt.Sprintf("Checking for DeploymentConfigs in project '%s' with selector '%s'", project, selector))
+	dcList, err := adapter.deploymentconfigs(project, kapi.ListOptions{LabelSelector: selector})
+	if err != nil {
+		r.Error("AGL0045", err, fmt.Sprintf("There was an error while trying to retrieve the DeploymentConfigs in project '%s': %s", project, err))
+		return
+	}
+	if len(dcList.Items) == 0 {
+		r.Error("AGL0047", nil, fmt.Sprintf("Did not find any matching DeploymentConfigs in project '%s'", project))
```

Comment:
A little context for what this implies would be good. Something like "This means that no logging components have been deployed."